src/axes.js

```diff
@@ -33,12 +34,22 @@ export function autoAxisTicks({x, y, fx, fy}, {x: xAxis, y: yAxis, fx: fxAxis, f
 }
 
 function autoAxisTicksK(scale, axis, k) {
+  tickInterval(scale, axis);
   if (axis.ticks === undefined) {
     const [min, max] = extent(scale.scale.range());
     axis.ticks = (max - min) / k;
   }
 }
 
+function tickInterval(scale, axis) {
+  const interval = maybeInterval(scale.interval);
+  if (interval != null) {
+    const [min, max] = extent(scale.scale.domain());
+    if (axis.ticks === undefined) axis.ticks = interval.range(interval.floor(min), interval.offset(interval.floor(max)));
+    if (scale.type !== "point" && scale.type !== "band" && axis.tickFormat === undefined && typeof scale.interval === "number") axis.tickFormat = ",";
```

Comment:
This should be using the isBandScale helper probably. Or maybe isOrdinalScale?

More to the point, what is this doing? We shouldn’t assume that a numeric interval is an integer; you can say interval: 0.5 (though maybe fractional intervals are not really recommended—it would probably be better for us to use interval: -2 like we do for d3.tickIncrement). In any case, rather than this default, maybe we should simply default tickFormat to [formatDefault](https://github.com/observablehq/plot/blob/93acd8220d4571506ace9d763847cfe6fad08669/src/format.js#L36) so that it works with number and date ticks?